docs/dns-health-checks.md

```diff
@@ -0,0 +1,86 @@
+# DNS Health Checks
+DNS Health Checks are a crucial tool for ensuring the availability and reliability of your multi-cluster applications. Kuadrant offers a powerful feature known as DNSPolicy, which allows you to configure and verify health checks for DNS endpoints. This guide provides a comprehensive overview of how to set up, utilize, and understand DNS health checks.
+
+## What are DNS Health Checks?
+DNS Health Checks are a way to assess the availability and health of DNS endpoints associated with your applications. These checks involve sending periodic requests to the specified endpoints to determine their responsiveness and health status. by configuring these checks via the DNS Policy, you can ensure that your applications are correctly registered, operational, and serving traffic as expected.
+
+## Configuration of Health Checks
+To configure a DNS health check, you need to deploy the provided YAML file. The key part of this configuration is the ‘healthCheck’ section, which includes important properties such as:
+
+* `allowInsecureCertificates`: Added for development environments, allows health probes to not fail when finding an invalid (e.g. self-signed) certificate.
+* `additionalHeadersRef`: This refers to a secret that holds extra headers, often containing important elements like authentication tokens.
+* `endpoint`: This is the path where the health checks take place, usually represented as '/healthz' or something similar.
+* `expectedResponses`: This setting lets you specify the expected HTTP response codes. If you don't set this, the default values assumed are 200 and 201.
+* `failureThreshold`: It's the number of times the health check can fail for the endpoint before it's marked as unhealthy.
+* `port`: Specific port for the connection to be checked.
+* `protocol`: Type of protocol being used, like HTTP or HTTPS.
+
+
+```
+kubectl apply -f - <<EOF
+apiVersion: kuadrant.io/v1alpha1
+kind: DNSPolicy
+metadata:
+  name: prod-web
+  namespace: multi-cluster-gateways
+spec:
+  targetRef:
+    name: prod-web
+    group: gateway.networking.k8s.io
+    kind: Gateway
+  healthCheck:
+    allowInsecureCertificates: true
+    endpoint: /
+    expectedResponses:
+      - 200
+      - 201
+      - 301
+    failureThreshold: 5
+    port: 443
+    protocol: https
+EOF
+```
+This configuration sets up a DNS health check for the specified prod-web Gateway using the defined parameters.
+
+## How to Validate DNS Health Checks
+
+After setting up DNS Health Checks to improve application reliability, it is important to verify their effectiveness. This guide provides a simple validation process to ensure that health checks are working properly and improving the operation of your applications.
+
+1. Verify Configuration:
+The first step in the validation process is to verify that the probes were created.
+```
+kubectl get dnshealthcheckprobes -A
```

Comment:
(From the top of my head) I think the probes have labels referencing the associated Gateway, so maybe we could add a label selector so it doesn't show all the probes in the cluster but the ones for the Gateway we're working with